apps/desktop/src/services/BuiltInToolService/tools/widgetTool/showWidget/runtime.ts

```
@@ -516,12 +517,17 @@ function parseRenderTree(rawHtml: string, phase: ShowWidgetPhase = 'ready'): Par
     const isFullDocument = /<(?:!doctype|html|head|body)\b/i.test(normalizedHtml);
 
+    // Allow <script> tags so runInlineScripts can re-execute them after morphdom
+    // patching. DOMPurify still strips event-handler attributes (onclick, onerror,
+    // etc.) and javascript: URIs, which is the actual XSS vector we need to block.
+    const purifyConfig: DOMPurify.Config = { ADD_TAGS: ['script'] };
+
     if (isFullDocument) {
         const parser = new DOMParser();
         const parsed = parser.parseFromString(normalizedHtml, 'text/html');
-        template.innerHTML = parsed.body.innerHTML || '<div></div>';
+        template.innerHTML = DOMPurify.sanitize(parsed.body.innerHTML || '<div></div>', purifyConfig);
```

ESLint

[error] Replace `parsed.body.innerHTML·||·'<div></div>',·purifyConfig` with `⏎············parsed.body.innerHTML·||·'<div></div>',⏎············purifyConfig⏎········`

(prettier/prettier)